FInal Project.R

```diff
@@ -1,66 +1,94 @@
+                           "single_epithelial_cell_size",
+                           "bare_nuclei",
+                           "bland_chromatin",
+                           "normal_nucleoli",
+                           "mitoses",
+                           "class")
+
+# Reconfirm that feature names were updated and preview data
+str(cancer_data)
+head(cancer_data)
 
 #Checking for missing variables
-sapply(df, function(x) sum(is.na(x)))
-
-
-df$bare_nuclei
-
-head(df$bare_nuclei)
-
-summary(df)
-
-#Visualization
-
-df2 <- df
-df2
+sapply(cancer_data, function(x) sum(is.na(x)))
 
-df2$bare_nuclei[ddf$bare_nuclei == "?"] <- NA
-df2$bare_neuclei
+# Drop "sample_code_number" as it is not noteworthy
+cancer_data <- select(cancer_data, -c("sample_code_number"))
 
-# Impute missing values with mean (you can choose a different imputation method)
-df2$bare_nuclei <- as.integer(ifelse(is.na(df2$bare_nuclei), median(df2$bare_nuclei, na.rm = TRUE), df2$bare_nuclei))
-df2$bare_nuclei
+str(cancer_data)
+head(cancer_data)
 
-
-#Data Exploration
-plot(df)
-
-hist(df2$clump_thickness)
-hist(df2$uniformity_of_cell_size)
-hist(df2$uniformity_of_cell_shape)
-hist(df2$marginal_adhesion)
-hist(df2$single_epithelial_cell_size)
-hist(df2$bare_nuclei)
-hist(df2$bland_chromatin)
-hist(df2$normal_nucleoli)
-hist(df2$mitoses)
-
-#code start
-hist(df2$class, breaks = seq(min(df2$class), max(df2$class), length.out = 1), axes = FALSE)
-axis(side=1, at = c(2.5,3.5), c("Benign", "Malignant"))
-box()
-axis(side = 2)
-#code end
-
-
-     
\ No newline at end of file
+#Checking for missing variables
+sapply(cancer_data, function(x) sum(is.na(x)))
+
+# Replace "?" `bare_nuclei` values with NA in the dataset;
+# This is missing data
+na_if(cancer_data$bare_nuclei, "?")
+
+# Impute NA values with median (for now)
+cancer_data$bare_nuclei[is.na(cancer_data$bare_nuclei)] <- median(cancer_data$bare_nuclei, na.rm = TRUE)
+
+# Data Exploration.
+# TODO: Generate Histograms for each feature, plus boxplots vs. the target classes
+
+# Boxplots
+ggplot(cancer_data) +
+  geom_boxplot(aes(x = factor(class, labels = c("Benign", "Malignant")),
+                   y = clump_thickness,
+                   fill = factor(class, labels = c("Benign", "Malignant")))) +
+  scale_x_discrete(labels = c('Benign', 'Malignant')) +
+  labs(x = "Tumour type",
+       y = "Clump thickness",
+       fill = "Legend",
+       title = "Boxplot of Clump thickness vs. classification",
+       caption = "Notice the clear differentiation in the range of clump thickness values") +
+  scale_fill_manual(values = c('Benign' = '#006989', 'Malignant' = '#CFDBD5'))
+
+# Histograms
+ggplot(cancer_data) +
+  geom_histogram(aes(x = clump_thickness),
+                 binwidth = 1,
+                 fill = "#006989",
+                 colour = "#1E1E1E") +
+  labs(x = "Clump thickness",
+       y = "Frequency",
+       title = "Histogram of clump thickness values")
+
+
+# hist(df2$clump_thickness)
```

Comment:
Commented for now. Wanted to discuss what diagrams we should be plotting + using `ggplot` to plot graphs.